Cogs/self_timeout.py

```diff
@@ -0,0 +1,41 @@
+from discord.ext import commands
+from datetime import timedelta
+import logging
+from .base_cog import BaseCog
+
+log = logging.getLogger(__name__)
+
+class SelfTimeout(BaseCog):
+    def __init__(self, bot):
+        BaseCog.__init__(self, bot)
+        self.bot = bot
+    
+    @commands.command(aliases=["selftimeout", "to"])
+    async def selfto(self, ctx, duration: str = commands.parameter(description="Length of time out. Examples: 2m = 2 minutes, 5h = 5 hours.")):
+        """Times the user out for a specified amount of time which cannot exceed 48 hours. Go do your work!"""
+        BaseCog.check_not_private(self, ctx)
+        if duration[-1].lower() not in ["h", "m"] or len(duration) <= 1:
```

Comment:
Shouldn't you be checking for the empty string before indexing?
I.e `if len(duration) <= 1 or duration[-1].lower() not in ["h", "m"]:`

I haven't tested but I assume that an empty string would throw an exception in the current code version.